### PR TITLE
Load appointments from db

### PR DESCRIPTION
**Pull request by tbonelaforge**

### CODE REVIEW

File: DrDocket/Appointment.h [hunk 1]
```diff
@@ -33,8 +36,47 @@ class Appointment {
...
         return result;
     }
-    
- private:
+
+    static void parseDatetime(string datetime, Date& date, Time& time) {
+
+        // Use format: 2016-01-01 10:20
+        replace(datetime.begin(), datetime.end(), '-', ' ');
+        std::replace(datetime.begin(), datetime.end(), ':', ' ');
```
Comment (AEGarten) [inline DrDocket/Appointment.h:44]: Why is this happening twice? Do you still need the std:: ?

File: DrDocket/Date.cpp [hunk 2]
```diff
@@ -46,13 +46,20 @@ Date
 #include <iostream>
 #include "Date.h"
 
-Date::Date(Month m, unsigned int d, unsigned int y) {
+Month Months[13] = {JAN, JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC};
```
Comment (AEGarten) [inline DrDocket/Date.cpp:49]: Why are there two Jan's?

File: DrDocket/Date.cpp [hunk 3]
```diff
@@ -46,13 +46,20 @@ Date
...
     month = m;
     day = (d < 32) ? d : 1;
     year = (y < 2100) ? y : 2019;
     weekNum = findWeekNum(m, day);
     weekday = findWeekDay(m, day);
 }
+
+Date::Date(Month m, unsigned int d, unsigned int y) {
+    this->initialize(m, d, y);
+}
```
Comment (AEGarten) [inline DrDocket/Date.cpp:61]: Why did the constructor need to change? It's not a problem, just curious.

File: DrDocket/Date.h [hunk 4]
```diff
@@ -54,6 +58,7 @@ class Date {
     bool operator< (const Date &rhs);
     bool operator<= (const Date &rhs);
     Date operator+ (const unsigned int &rhs);    //increment Date by days
+    static Month parseMonth(int month);
```
Comment (AEGarten) [inline DrDocket/Date.h:61]: Does this need to be static?

File: DrDocket/Resource.cpp [hunk 5]
```diff
@@ -274,13 +269,7 @@ void Resource::convertAvailabilityToAppointment(Appointment availability, Appoin
...
     addAppt(appointment, 0);
     if (type == "Dr") { // commit to paperwork
-        Appointment admin = Appointment(
-                "Admin",
-                appointment.getStart() + appointment.getDuration(),
-                Time(0, 5),
-                availability.getDay()
-        );
-        addAppt(admin, 0); //paperwork for doc
+        addAdminAppointment(appointment);
```
Comment (AEGarten) [inline DrDocket/Resource.cpp:272]: Nice!

File: DrDocket/Spider.cpp [hunk 6]
```diff
@@ -1,4 +1,5 @@
 #include "Spider.h"
+#include "Appointment.h"
```
Comment (AEGarten) [inline DrDocket/Spider.cpp:2]: Spider header includes the Resource header which includes the Appointment header. A comment on this dependency could have been left by me though.

File: DrDocket/Spider.cpp [hunk 7]
```diff
@@ -167,34 +243,12 @@ bool Spider::findMatchTime(Resource* rm, Opens &opens) {
...
-            break;
-            
-        case XRAY:
-            dur.setMn(20);
-            type = "X-Ray";
-            break;
-            
-        case NIL: throw "Invalid Requirement for Appointment";
-    }
+    Appointment::parseRequirement(req, dur, type);
```
Comment (AEGarten) [inline DrDocket/Spider.cpp:251]: Fair

File: DrDocket/Spider.cpp [hunk 8]
```diff
@@ -69,6 +80,82 @@ int insertAvailabilityCallback(void * NotUsed, int argc, char ** argv, char ** c
...
+    return 0;
+}
+
+
+void prepareGetAppointmentsByResourceQuery(string resource_type, int resource_id) {
+    GET_APPOINTMENTS_BY_RESOURCE_QUERY[0] = '\n';
+    sprintf(
+            GET_APPOINTMENTS_BY_RESOURCE_QUERY,
+            GET_APPOINTMENTS_BY_RESOURCE_TEMPLATE,
+            (resource_type == "Dr") ? "doctor_id" : (resource_type == "Room") ? "room_id" : "patient_id",
```
Comment (AEGarten) [inline DrDocket/Spider.cpp:123]: Well I just learned that you can chain the ternery operator.

**Review by AEGarten — COMMENTED** [review]
See line comments in files changed.

File: DrDocket/Date.cpp (hunk 2 above)
Comment (tbonelaforge) [reply DrDocket/Date.cpp:49]: To facilitate looking up the proper enum for months 1, 2, ..., 12, 1 => JAN, 2 => FEB.
With this array you can do properEnum = Months[month]

File: DrDocket/Date.cpp (hunk 3 above)
Comment (tbonelaforge) [reply DrDocket/Date.cpp:61]: I was working on the "parseDatetime" function, (https://github.com/tbonelaforge/appointments-lite/blob/load-appointments-from-db/DrDocket/Appointment.h#L40) which uses date and time objects passed by reference, as a place to store the parsed results.
In that case I needed to call "initialize" on an existing instance, resulting in the same object that the original constructor would have produced.
Factored out the entire body of the constructor so I could use it in two places!

File: DrDocket/Appointment.h (hunk 1 above)
Comment (tbonelaforge) [reply DrDocket/Appointment.h:44]: The first one is replacing dashes, the second is replacing colons.
Maybe there's a way to do them both in one pass (possibly using RegEx) but I'm not too worried about optimizing this part.

File: DrDocket/Date.h (hunk 4 above)
Comment (tbonelaforge) [reply DrDocket/Date.h:61]: It doesn't need to, but I thought it was a clearer interface to the code.  I.e This function can be used without having to construct a Date instance first.

**Pull request merged**